### DISCUSSION TITLE
"Attempting to connect to manager" message at top of console in standalone mode

### DISCUSSION THREAD

**Discussion by jaredatobe** [Security-Onion-Solutions/securityonion#4734] [Unsupported Versions] [answered]
After installation, while reviewing the freshly-installed Security Onion console, I see a message at the top. It toggles between "Attempting to connect to manager" and "Disconnected from manager" continuously, every few seconds. It's a standalone install.. So-status shows that everything it sees is OK. Where should I look to begin resolving this?

**Comment by jertel** [comment]
The web page is attempting to open a websocket to the server, but something is preventing that connection. You can use developer tools in your browser to view the console errors and network traffic.

**Reply by jaredatobe** [reply]
Thank you, this got me pointed in the right direction. I posted the fix below.

**Comment by jaredatobe** [comment, marked as answer]
Thank you for the tip! I'm using a reverse proxy in front of security onion, so I had to add a websocket stanza to the Apache configuration:

This part already existed in the Virtual Host config:

```
ProxyPreserveHost on
ProxyRequests off
ProxyPass "/" https://example.com/
ProxyPassReverse "/" https://example.com/
```

I added this to get websockets working:

```
RewriteEngine on
RewriteCond %{HTTP:Upgrade} websocket [NC]
RewriteCond %{HTTP:Connection} upgrade [NC]
RewriteRule ^/?(.*) "wss://example.com/$1" [P,L]
```
Note that you'll need to run "sudo a2enmod proxy_wstunnel" at the command line in order to enable the wstunnel module which interprets this code. If you add the code above but do not enable the wstunnel module, you'll still encounter the issue.

If you want to dig deeper into this, there are lots of how-tos. Here's a good place to start: [ServerLab: How to Reverse Proxy Websockets with Apache 2.4]( https://www.serverlab.ca/tutorials/linux/web-servers-linux/how-to-reverse-proxy-websockets-with-apache-2-4/)
 
Thanks
